python/ray/serve/_private/proxy.py

```diff
@@ -668,8 +677,16 @@ async def unary_stream(
                 else:
                     yield message
 
-            context.set_code(status.code)
-            context.set_details(status.message)
+            # Only the latest code and details will take effect. If the user already
+            # set them to a truthy value in the context, skip setting them with Serve's
+            # default values. By default, if nothing is set, the code is 0 and the
+            # details is "", which both are falsy. So if the user did not set them or
+            # if they're explicitly set to falsy values, such as None, Serve will
+            # continue to set them with our default values.
+            if not context.code():
+                context.set_code(status.code)
+            if not context.details():
+                context.set_details(status.message)
```

Comment:
possible to refactor to avoid duplicating this logic & comment? (ok if it's hard)